compiler/rustc_typeck/src/hir_wf_check.rs

```diff
@@ -141,6 +141,11 @@ fn diagnostic_hir_wf_check<'tcx>(
                 ref item => bug!("Unexpected item {:?}", item),
             },
             hir::Node::Field(field) => Some(field.ty),
+            hir::Node::ForeignItem(item) => match item.kind {
+                ForeignItemKind::Fn(_, _, _) => None,
+                ForeignItemKind::Static(ty, _) => Some(ty),
+                ForeignItemKind::Type => None,
```

Comment:
```suggestion
                ref item => bug!("Unexpected foreign item {:?}", item),
```
Both `ForeignItemKind::Fn` and `ForeignItemKind::Type` should be unreachable here.

Reply:
@petrochenkov Sorry, I missed your suggestion and ended up constraining the match instead. Is it better to match ForeignType and then match on its kind?

Reply:
The current version is ok too.